File: 1468-check-if-n-and-its-double-exist/check-if-n-and-its-double-exist.py
class Solution:
    def checkIfExist(self, arr: List[int]) -> bool:
        visited = set()
        for i in range(len(arr)):
            if arr[i]*2 in visited or (arr[i]%2==0 and arr[i]//2 in visited):
                return True
            visited.add(arr[i])
        return False
        

File: 1468-check-if-n-and-its-double-exist/check-if-n-and-its-double-exist.java
class Solution {
    public boolean checkIfExist(int[] arr) {
        Set<Integer> visited = new HashSet<>();
        for (int i=0;i<arr.length;i++){
            if ( visited.contains(arr[i]*2)|| ((arr[i]%2==0) && ( visited.contains(arr[i]/2))))
                        {
                return true;
            
            }
            visited.add(arr[i]);
        }
        return false;
    }
}

